server/app/jobs/grid_service_scheduler_worker.rb

```diff
@@ -41,6 +41,10 @@ def check_deploy_queue
         return nil
       end
     end
+  rescue => exc
+    warn "aborting deploy on un-handled error: #{exc.message}"
+    warn exc.backtrace.join("\n") if exc.backtrace
+    service_deploy.abort! "service deploy aborted: #{exc.message}" if service_deploy
```

Comment:
Currently the CLI considers the aborted deploy to be a success, because it's in the `created` state, not the `error` state :/

```
    *   Deploying service redis-ha-stateful  .. \[API Client Response]: Headers: {Content-Type: application/json}  | Status: 200 | Body: 
{"id":"591074c8de35782c7f000000","created_at":"2017-05-08T13:38:16.300Z","queued_at":"2017-05-08T13:38:16.386+00:00","started_at":null,"finished_at":"2017-05-08T13:38:16.392+00:00","service_id":"development/null/redis-ha-stateful","state":"created","reason":"service deploy aborted: testing","instance_count":1,"instance_deploys":[]}
 [done] Deploying service redis-ha-stateful      
```

This is an omission in #2221... needs CLI changes to fix.